Added cpp versions of sysid routines and example

r4stered opened the pull request.

Oblarg commented on wpilibcExamples/src/main/cpp/examples/SysId/include/subsystems/Drive.h:
```diff
@@ -0,0 +1,78 @@
+          },
+          [this](frc::sysid::MotorLog* log) {
+            log->RecordFrameLinear(
+                m_appliedVoltage = m_leftMotors.Get() *
```
Should add a `getVoltage` method to the CPP-side `MotorController` class that returns in the right units

calcmogul replied:
That's out of scope for this PR, so I opened an issue: https://github.com/wpilibsuite/allwpilib/issues/6043

calcmogul committed:
[wpimath] Add more docs to SimulatedAnnealing (NFC) (#6054)

Oblarg committed:
[commands] SubsystemBase: allow setting name in constructor (#6052)

calcmogul commented:
Looks like you messed up your rebase.

r4stered commented:
Yeah, I got my repo in a bad state and ended up having to reclone... I am working on adding the unit tests right now. Should be done today hopefully.

Commit:
Add cpp version of sysid routines and ex project

r4stered commented:
Should have unit tests now!

Commit:
Add cpp version of sysid routines and ex project

PeterJohnson committed:
[glass] Check for null entries when updating struct/proto (#6059)

PeterJohnson committed:
[ntcore] Fix overlapping subscriber handling (#6067)

calcmogul committed:
[sysid] Document timestamp units (#6065)

spacey-sooty committed:
[build] Upgrade quickbuf to 1.3.3 (#6072)

sciencewhiz committed:
[wpilib] AddressableLED: Update warning about single driver (NFC) (#6069)

Say that multiple strips can be used in series.

ThadHouse committed:
[cscore] Fix RawFrame class not being loaded in JNI (#6077)

calcmogul committed:
[wpilib] PS4Controller: enable usage reporting (#6081)

rzblue committed:
[dlt] Change datalogtool default folder to logs folder (#6079)

Also fix straggling documentation.

rzblue committed:
[hal,wpilib] Add function to control "Radio" LED (#6073)

r4stered committed:
[apriltag] Fix Java generation functions (#6063)

m10653 committed:
[glass] Add Profiled PID controller support & IZone Support (#5959)

crueter committed:
[build] cmake: Add NO_WERROR option to disable -Werror (#6071)

ncorrea210 committed:
[wpimath] Make feedforward classes throw exceptions for negative Kv or Ka (#6084)

ncgears committed:
[wpilib] Add StadiaController and command wrapper (#6083)

PeterJohnson committed:
[wpiutil] Add serializable marker interfaces (#6060)

rzblue committed:
[commands] Fix compose-while-scheduled issue and test all compositions (#5581)

tandej committed:
[wpilib] Add method to get breaker fault at a specific channel in PowerDistribution[Sticky]Faults (#5521)

Co-authored-by: Ryan Blue <ryanzblue@gmail.com>

calcmogul committed:
[wpimath] Report error on negative PID gains (#6055)

Defaults PID gains to zero if any are invalid.

truher committed:
[glass] Fix order of loading window settings (#6056)

PeterJohnson committed:
[wpiutil] DataLog: Fix UB in AppendImpl (#6088)

PeterJohnson committed:
[wpiutil] DataLog: Ensure file is written on shutdown (#6087)

Previously the thread could end without the file being written.

sciencewhiz committed:
[wpilib] Add PWMSparkFlex MotorController (#6089)

auscompgeek committed:
[ci] Update artifact actions to v4 (#6092)

auscompgeek committed:
[ci] Update pregenerate workflow to actions/checkout@v4 (#6094)

KangarooKoala committed:
[commands] Deprecate C++ TransferOwnership() (#6095)

It has been completely replaced with ToPtr().

calcmogul committed:
[wpilib] Fix precision issue in Color round-and-clamp (#6100)

bryceroethel committed:
[wpilib] ADIS16470: Add access to all 3 axes (#6074)

PeterJohnson committed:
[wpiutil,cscore] Move VideoMode.PixelFormat to wpiutil (#6097)

This tracks the RawFrame move.

calcmogul commented:
The C++ tests are crashing.

KangarooKoala committed:
[commands] Improve error message when composing commands twice in same composition (#6091)

Also disallow deadline command from also appearing in other commands.

PeterJohnson committed:
[wpiutil,cscore,apriltag] Fix RawFrame (#6098)

calcmogul committed:
[sysid] Fix adjusted R² calculation (#6101)

It hardcoded p to 2.

Gold856 committed:
[wpilib] Fix usage reporting for static classes (#6090)

r4stered commented:
> The C++ tests are crashing.

Thats odd. Once I get some time I can try to reproduce the crash. Been busy with work lately....

calcmogul committed:
[commands] Mark ParallelDeadlineGroup.setDeadline() final (#6102)

Suppress this-escape warning.

PeterJohnson committed:
[wpiutil] Struct: Add info template parameter pack (#6086)

This allows using Struct in a dynamically typed environment by passing
additional information to the Struct serialization functions.

r4stered committed:
clean up commit mess

r4stered committed:
Merge branch 'wpilibsuite:main' into sysid-routines-cpp
